### ISSUE TITLE
move validation to parity-wasm

### ISSUE THREAD

**Issue by pchickey** [paritytech/wasmi#111]
I'm a user of `parity-wasm` but not of `wasmi`. When I load wasm from an untrusted source, or I manipulate a wasm AST, I'd like to ensure it is valid. There is a validator in https://github.com/paritytech/wasmi/blob/master/src/validation/mod.rs#L168 that I would like to use without adding a dependency on `wasmi`. Are you open to moving that code to the `parity-wasm` crate, along with the ModuleContext machinery required to make it work? I'm sorry if this questions a design decision you already made - its OK if you don't feel its appropriate to move this code.

**Comment by pepyakin** [comment]
The validator is used to be inside `parity-wasm`. It was moved simply because it was essential for `wasmi` and at that time it was already obvious that it would be more of loading wasm module rather than just validating it. And recently we started to use this tight coupling for implementing [compilation](https://github.com/paritytech/wasmi/blob/a605175abe3e3578bdda2d91e565959dab908df2/src/validation/func.rs#L394) of wasm into a [format](https://github.com/paritytech/wasmi/blob/master/src/isa.rs) which is more convenient for interpretation. So it might be a little complicated to move it as is.

That said, I still think it may be useful to have validation in `parity-wasm`. It was even considered to just have a copy of the validation code if there is a great need.
